image/preseed/preseed_linux.go

```diff
@@ -233,13 +237,15 @@ func prepareCore20Mountpoints(prepareImageDir, tmpPreseedChrootDir, snapdSnapBlo
 		}
 		// cleanup after handle-writable-paths
 		for _, ent := range entries {
-			if strings.HasPrefix(ent.MountDir, tmpPreseedChrootDir) {
-				cmd := exec.Command("umount", ent.MountDir)
-				if out, err := cmd.CombinedOutput(); err != nil {
-					fmt.Fprintf(Stdout, "cannot unmount: %v\n'umount %s' failed with: %s", err, ent.MountDir, out)
-				}
+			if ent.MountDir != tmpPreseedChrootDir && strings.HasPrefix(ent.MountDir, tmpPreseedChrootDir) {
+				umountCmd(ent.MountDir)
 			}
 		}
+
+		// finally, umount the base snap
```

Comment:
(nitpick^3) I wonder if we should error here if len(moutned) > 1 ? so something like:
```
switch len(mounted) {
  case 1: umountCmd(mounted[0])
  default: return fmr.Errorf("internal error: unexpected number of mounted filesystems %v: %v", len(mounted), mounted)
 }
  ```

Reply:
We actually expect more than 1 mount entry here (unless the code gets modified to drop them from the array as we go over them in the loop above); this is just special-casing entry #0 as it's the base.